### ISSUE TITLE
Investigate why creating rand matrices is very slow

### ISSUE THREAD

**Issue by sadikovi** [sadikovi/rustjblas#10]
Creating random matrices of 4000x4000 takes about 32 seconds, meanwhile it takes 1 second in jblas.

**Comment by sadikovi** [comment]
Bench scripts:
```scala
def func(n: Int = 100): Unit = {
  val start1 = System.nanoTime()
  val a = org.jblas.DoubleMatrix.rand(n, n)
  val b = org.jblas.DoubleMatrix.rand(n, n)
  val end1 = System.nanoTime()
  println(s"create matrices: ${(end1 - start1)/1e6} ms")
  
  val start = System.nanoTime()
  a.mmul(b)
  val end = System.nanoTime()
  println(s"n: $n, took ${(end - start)/1e6} ms")
}

def func(n: Int = 100): Unit = {
  val start1 = System.nanoTime()
  val a = com.github.sadikovi.rustjblas.DoubleMatrix.rand(n, n)
  val b = com.github.sadikovi.rustjblas.DoubleMatrix.rand(n, n)
  val end1 = System.nanoTime()
  println(s"create matrices: ${(end1 - start1)/1e6} ms")
  
  val start = System.nanoTime()
  a.mmul(b)
  val end = System.nanoTime()
  a.dealloc
  b.dealloc
  println(s"n: $n, took ${(end - start)/1e6} ms")
}
```

**Comment by sadikovi** [comment]
Current benchmakrs on ubuntu 14.04:
```
// jblas
scala> func(4000)
create matrices: 804.718473 ms
matrix multiply, n: 4000, took 10768.599228 ms

// rustjblas
scala> func(4000)
create matrices: 36906.304934 ms // I am working on it, it should not take that long...
matrix multiply, n: 4000, took 978.402747 ms
```

**Comment by sadikovi** [comment]
I replaced random number generation for matrices. Performance numbers (with release flag on):

jblas
```scala
scala> func(1000)
create matrices: 79.134667 ms
matrix multiply, n: 1000, took 228.864138 ms

scala> func(4000)
create matrices: 734.102912 ms
matrix multiply, n: 4000, took 10263.439831 ms

scala> func(8000)
create matrices: 3205.571456 ms
matrix multiply, n: 8000, took 83460.120528 ms
```

rustjblas
```java
scala> func(1000)
-- Loading library rustjblas, libpath: target
create matrices: 25.756595 ms
matrix multiply, n: 1000, took 31.747341 ms

scala> func(4000)
create matrices: 192.274154 ms
matrix multiply, n: 4000, took 993.17971 ms

scala> func(8000)
create matrices: 1172.633004 ms
matrix multiply, n: 8000, took 7826.578085 ms
```

**Comment by sadikovi** [comment]
Fixed in b9a1641447d08159bc3a7ca9b27d797cbdb3c759
